src/core/src/device/dummy.rs

```diff
@@ -0,0 +1,185 @@
+#![allow(missing_docs)]
+use device as d;
+use device::{Device, Resources, Capabilities, SubmitInfo};
+use device::draw::{CommandBuffer, Access, Gamma, Target};
+use device::shade;
+use super::{tex};
+use draw_state::target::{Rect, Mirror, Mask, ClearData, Layer, Level};
+
+pub struct DummyDevice {
+    capabilities: Capabilities
+}
+pub struct DummyCommandBuffer {
+    buf: Vec<String>
```

Comment:
Rather then make this an buffer of strings, it should probably be a buffer of `Commands`. The `Commands` can easily be turned back into a string using `std::fmt::Debug`

We also need some way of reading this Vec back.